// solidity-antlr4/dist/ast/yul/index.node.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "YulAssignment", {
  enumerable: true,
  get: function () {
    return _yulAssignment.YulAssignment;
  }
});
Object.defineProperty(exports, "YulBlock", {
  enumerable: true,
  get: function () {
    return _yulBlock.YulBlock;
  }
});
Object.defineProperty(exports, "YulBoolean", {
  enumerable: true,
  get: function () {
    return _yulBoolean.YulBoolean;
  }
});
Object.defineProperty(exports, "YulForStatement", {
  enumerable: true,
  get: function () {
    return _yulForStatement.YulForStatement;
  }
});
Object.defineProperty(exports, "YulFunctionCall", {
  enumerable: true,
  get: function () {
    return _yulFunctionCall.YulFunctionCall;
  }
});
Object.defineProperty(exports, "YulFunctionDefinition", {
  enumerable: true,
  get: function () {
    return _yulFunctionDefinition.YulFunctionDefinition;
  }
});
Object.defineProperty(exports, "YulIfStatement", {
  enumerable: true,
  get: function () {
    return _yulIfStatement.YulIfStatement;
  }
});
Object.defineProperty(exports, "YulLiteral", {
  enumerable: true,
  get: function () {
    return _yulLiteral.YulLiteral;
  }
});
Object.defineProperty(exports, "YulPath", {
  enumerable: true,
  get: function () {
    return _yulPath.YulPath;
  }
});
Object.defineProperty(exports, "YulStatement", {
  enumerable: true,
  get: function () {
    return _yulStatement.YulStatement;
  }
});
Object.defineProperty(exports, "YulSwitchCase", {
  enumerable: true,
  get: function () {
    return _yulSwitchCase.YulSwitchCase;
  }
});
Object.defineProperty(exports, "YulSwitchStatement", {
  enumerable: true,
  get: function () {
    return _yulSwitchStatement.YulSwitchStatement;
  }
});
Object.defineProperty(exports, "YulVariableDeclaration", {
  enumerable: true,
  get: function () {
    return _yulVariableDeclaration.YulVariableDeclaration;
  }
});
var _yulAssignment = require("./yul-assignment.cjs");
var _yulBlock = require("./yul-block.cjs");
var _yulBoolean = require("./yul-boolean.cjs");
var _yulForStatement = require("./yul-for-statement.cjs");
var _yulFunctionCall = require("./yul-function-call.cjs");
var _yulFunctionDefinition = require("./yul-function-definition.cjs");
var _yulIfStatement = require("./yul-if-statement.cjs");
var _yulLiteral = require("./yul-literal.cjs");
var _yulPath = require("./yul-path.cjs");
var _yulStatement = require("./yul-statement.cjs");
var _yulSwitchCase = require("./yul-switch-case.cjs");
var _yulSwitchStatement = require("./yul-switch-statement.cjs");
var _yulVariableDeclaration = require("./yul-variable-declaration.cjs");
