// solidity-antlr4/dist/ast/declaration/index.node.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ContractDefinition", {
  enumerable: true,
  get: function () {
    return _contractDefinition.ContractDefinition;
  }
});
Object.defineProperty(exports, "EnumDefinition", {
  enumerable: true,
  get: function () {
    return _enumDefinition.EnumDefinition;
  }
});
Object.defineProperty(exports, "ErrorDefinition", {
  enumerable: true,
  get: function () {
    return _errorDefinition.ErrorDefinition;
  }
});
Object.defineProperty(exports, "EventDefinition", {
  enumerable: true,
  get: function () {
    return _eventDefinition.EventDefinition;
  }
});
Object.defineProperty(exports, "FunctionDefinition", {
  enumerable: true,
  get: function () {
    return _functionDefinition.FunctionDefinition;
  }
});
Object.defineProperty(exports, "ModifierDefinition", {
  enumerable: true,
  get: function () {
    return _modifierDefinition.ModifierDefinition;
  }
});
Object.defineProperty(exports, "StructDefinition", {
  enumerable: true,
  get: function () {
    return _structDefinition.StructDefinition;
  }
});
Object.defineProperty(exports, "StructMember", {
  enumerable: true,
  get: function () {
    return _structMember.StructMember;
  }
});
Object.defineProperty(exports, "UserDefinedValueTypeDefinition", {
  enumerable: true,
  get: function () {
    return _userDefinedValueTypeDefinition.UserDefinedValueTypeDefinition;
  }
});
Object.defineProperty(exports, "VariableDeclaration", {
  enumerable: true,
  get: function () {
    return _variableDeclaration.VariableDeclaration;
  }
});
var _contractDefinition = require("./contract-definition.cjs");
var _enumDefinition = require("./enum-definition.cjs");
var _errorDefinition = require("./error-definition.cjs");
var _eventDefinition = require("./event-definition.cjs");
var _functionDefinition = require("./function-definition.cjs");
var _modifierDefinition = require("./modifier-definition.cjs");
var _structDefinition = require("./struct-definition.cjs");
var _structMember = require("./struct-member.cjs");
var _userDefinedValueTypeDefinition = require("./user-defined-value-type-definition.cjs");
var _variableDeclaration = require("./variable-declaration.cjs");
