// solidity-antlr4/dist/ast/statement/index.node.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AssemblyStatement", {
  enumerable: true,
  get: function () {
    return _assemblyStatement.AssemblyStatement;
  }
});
Object.defineProperty(exports, "Block", {
  enumerable: true,
  get: function () {
    return _block.Block;
  }
});
Object.defineProperty(exports, "BreakStatement", {
  enumerable: true,
  get: function () {
    return _breakStatement.BreakStatement;
  }
});
Object.defineProperty(exports, "CatchClause", {
  enumerable: true,
  get: function () {
    return _catchClause.CatchClause;
  }
});
Object.defineProperty(exports, "ContinueStatement", {
  enumerable: true,
  get: function () {
    return _continueStatement.ContinueStatement;
  }
});
Object.defineProperty(exports, "DoWhileStatement", {
  enumerable: true,
  get: function () {
    return _doWhileStatement.DoWhileStatement;
  }
});
Object.defineProperty(exports, "EmitStatement", {
  enumerable: true,
  get: function () {
    return _emitStatement.EmitStatement;
  }
});
Object.defineProperty(exports, "ExpressionStatement", {
  enumerable: true,
  get: function () {
    return _expressionStatement.ExpressionStatement;
  }
});
Object.defineProperty(exports, "ForStatement", {
  enumerable: true,
  get: function () {
    return _forStatement.ForStatement;
  }
});
Object.defineProperty(exports, "IfStatement", {
  enumerable: true,
  get: function () {
    return _ifStatement.IfStatement;
  }
});
Object.defineProperty(exports, "ReturnStatement", {
  enumerable: true,
  get: function () {
    return _returnStatement.ReturnStatement;
  }
});
Object.defineProperty(exports, "RevertStatement", {
  enumerable: true,
  get: function () {
    return _revertStatement.RevertStatement;
  }
});
Object.defineProperty(exports, "TryStatement", {
  enumerable: true,
  get: function () {
    return _tryStatement.TryStatement;
  }
});
Object.defineProperty(exports, "VariableDeclarationStatement", {
  enumerable: true,
  get: function () {
    return _variableDeclarationStatement.VariableDeclarationStatement;
  }
});
Object.defineProperty(exports, "WhileStatement", {
  enumerable: true,
  get: function () {
    return _whileStatement.WhileStatement;
  }
});
var _assemblyStatement = require("./assembly-statement.cjs");
var _block = require("./block.cjs");
var _breakStatement = require("./break-statement.cjs");
var _catchClause = require("./catch-clause.cjs");
var _continueStatement = require("./continue-statement.cjs");
var _doWhileStatement = require("./do-while-statement.cjs");
var _emitStatement = require("./emit-statement.cjs");
var _expressionStatement = require("./expression-statement.cjs");
var _forStatement = require("./for-statement.cjs");
var _ifStatement = require("./if-statement.cjs");
var _returnStatement = require("./return-statement.cjs");
var _revertStatement = require("./revert-statement.cjs");
var _tryStatement = require("./try-statement.cjs");
var _variableDeclarationStatement = require("./variable-declaration-statement.cjs");
var _whileStatement = require("./while-statement.cjs");
