// solidity-antlr4/dist/ast/expression/binary-operation.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ShiftOperation = exports.OrderComparison = exports.OrOperation = exports.MulDivModOperation = exports.ExpOperation = exports.EqualityComparison = exports.BitXorOperation = exports.BitOrOperation = exports.BitAndOperation = exports.BinaryOperation = exports.AndOperation = exports.AddSubOperation = void 0;
var _base = require("../base.cjs");
class BinaryOperation extends _base.BaseNode {
  type = "BinaryOperation";
  operator = null;
  left = null;
  right = null;
  constructor(ctx, visitor) {
    super(ctx, visitor);
    this.left = ctx.getChild(0)?.accept(visitor) ?? null;
    this.operator = ctx.getChild(1)?.getText() ?? null;
    this.right = ctx.getChild(2)?.accept(visitor) ?? null;
  }
}
exports.OrOperation = exports.OrderComparison = exports.ShiftOperation = exports.MulDivModOperation = exports.ExpOperation = exports.EqualityComparison = exports.BitXorOperation = exports.BitOrOperation = exports.BitAndOperation = exports.AddSubOperation = exports.AndOperation = exports.BinaryOperation = BinaryOperation;
