// solidity-antlr4/dist/ast/meta/index.node.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "DataLocation", {
  enumerable: true,
  get: function () {
    return _dataLocation.DataLocation;
  }
});
Object.defineProperty(exports, "IdentifierPath", {
  enumerable: true,
  get: function () {
    return _identifierPath.IdentifierPath;
  }
});
Object.defineProperty(exports, "ImportAliases", {
  enumerable: true,
  get: function () {
    return _importAliases.ImportAliases;
  }
});
Object.defineProperty(exports, "ImportDirective", {
  enumerable: true,
  get: function () {
    return _importDirective.ImportDirective;
  }
});
Object.defineProperty(exports, "InheritanceSpecifier", {
  enumerable: true,
  get: function () {
    return _inheritanceSpecifier.InheritanceSpecifier;
  }
});
Object.defineProperty(exports, "ModifierInvocation", {
  enumerable: true,
  get: function () {
    return _modifierInvocation.ModifierInvocation;
  }
});
Object.defineProperty(exports, "Path", {
  enumerable: true,
  get: function () {
    return _path.Path;
  }
});
Object.defineProperty(exports, "PragmaDirective", {
  enumerable: true,
  get: function () {
    return _pragmaDirective.PragmaDirective;
  }
});
Object.defineProperty(exports, "SourceUnit", {
  enumerable: true,
  get: function () {
    return _sourceUnit.SourceUnit;
  }
});
Object.defineProperty(exports, "UsingAliases", {
  enumerable: true,
  get: function () {
    return _usingAliases.UsingAliases;
  }
});
Object.defineProperty(exports, "UsingDirective", {
  enumerable: true,
  get: function () {
    return _usingDirective.UsingDirective;
  }
});
var _dataLocation = require("./data-location.cjs");
var _identifierPath = require("./identifier-path.cjs");
var _importAliases = require("./import-aliases.cjs");
var _importDirective = require("./import-directive.cjs");
var _inheritanceSpecifier = require("./inheritance-specifier.cjs");
var _modifierInvocation = require("./modifier-invocation.cjs");
var _path = require("./path.cjs");
var _pragmaDirective = require("./pragma-directive.cjs");
var _sourceUnit = require("./source-unit.cjs");
var _usingAliases = require("./using-aliases.cjs");
var _usingDirective = require("./using-directive.cjs");
